__tests__/oas.test.js

```diff
@@ -3,280 +3,361 @@ const { Operation } = require('../src/oas');
     });
-  });
 
-  it('should work for petstore', () => {
-    const operation = new Oas(petstore).operation('/pet', 'post');
+    it('should work for multiple securities (&& and ||)', () => {
+      const operation = new Oas(multipleSecurities).operation('/and-or-security', 'post');
 
-    expect(operation.prepareSecurity()).toMatchSnapshot();
-  });
+      expect(operation.prepareSecurity().OAuth2).toHaveLength(2);
+      expect(operation.prepareSecurity().Header).toHaveLength(1);
+    });
 
-  it('should work for multiple securities (||)', () => {
-    const operation = new Oas(multipleSecurities).operation('/or-security', 'post');
+    it.todo('should set a `key` property');
 
-    expect(Object.keys(operation.prepareSecurity())).toHaveLength(2);
-  });
+    // TODO We dont currently support cookies?
+    it.todo('apiKey/cookie: should return with a type of Cookie');
 
-  it('should work for multiple securities (&&)', () => {
-    const operation = new Oas(multipleSecurities).operation('/and-security', 'post');
+    it.todo('should throw if attempting to use a non-existent scheme');
 
-    expect(Object.keys(operation.prepareSecurity())).toHaveLength(2);
-  });
+    it('should return empty object if no security', () => {
+      const operation = new Oas(multipleSecurities).operation('/no-auth', 'post');
+      expect(Object.keys(operation.prepareSecurity())).toHaveLength(0);
+    });
 
-  it('should work for multiple securities (&& and ||)', () => {
-    const operation = new Oas(multipleSecurities).operation('/and-or-security', 'post');
+    it('should return empty object if security scheme doesnt exist', () => {
+      const operation = new Oas(multipleSecurities).operation('/unknown-scheme', 'post');
+      expect(Object.keys(operation.prepareSecurity())).toHaveLength(0);
+    });
 
-    expect(operation.prepareSecurity().OAuth2).toHaveLength(2);
-    expect(operation.prepareSecurity().Header).toHaveLength(1);
+    it('should return empty if security scheme type doesnt exist', () => {
+      const operation = new Oas(multipleSecurities).operation('/unknown-auth-type', 'post');
+      expect(Object.keys(operation.prepareSecurity())).toHaveLength(0);
+    });
   });
 
-  it.todo('should set a `key` property');
+  describe('getHeaders()', () => {
+    it('should return an object containing request headers if params exist', () => {
+      const oas = new Oas(petstore);
+      const uri = `http://petstore.swagger.io/v2/pet/1`;
+      const method = 'DELETE';
 
-  // TODO We dont currently support cookies?
-  it.todo('apiKey/cookie: should return with a type of Cookie');
+      const logOperation = oas.findOperation(uri, method);
+      const operation = new Operation(oas, logOperation.url.path, logOperation.url.method, logOperation.operation);
 
-  it.todo('should throw if attempting to use a non-existent scheme');
+      expect(operation.getHeaders()).toMatchObject({
+        request: ['api_key'],
```

Comment:
This still returning `api_key` instead of `Authorization`?

Reply:
Correct. Per their docs, it looks like scheme types of `apiKey` come in as unique headers: https://swagger.io/docs/specification/authentication/api-keys/

Reply:
Shouldn't they still have an `in` that represents the location of where it's sent?

Reply:
Well we're only concerned with headers, and it's being filtered as such. If it's added as a querystring that's not something should be returned or omitted from our log view.

Reply:
I'm just confused why it's being returned as `api_key`. From my understanding of the FE implementation for this, we want to segment out headers in a response that aren't stored in their OAS, and to do that we'll need to know the name of the header (`Authorization` in this case). Pulling back `api_key` is going to require another lookup to see what `api_key` matches up to.

Reply:
We should be returning the name of the api key, not `api_key` itself. There shouldn't be a need to find out what it's attached to after being served. I believe in relation to the test, it's named `api_key` in the petstore.json fixture for the `DELETE` method. Would you like me to update that value so that it doesn't seem conflated?